File: jvm/core/main/scala/com/webank/eggroll/core/deepspeed/store/meta/RendezvousStoreDestroyResponse.scala
package com.webank.eggroll.core.deepspeed.store.meta

import com.google.protobuf.ByteString
import com.webank.eggroll.core.meta.Deepspeed

import scala.language.implicitConversions

case class RendezvousStoreDestroyResponse(success: Boolean)

object RendezvousStoreDestroyResponse {
  implicit def serialize(src: RendezvousStoreDestroyResponse): Array[Byte] = {
    val builder = Deepspeed.StoreDestroyResponse.newBuilder()
      .setSuccess(src.success)
    builder.build().toByteArray
  }

  implicit def deserialize(byteString: ByteString): RendezvousStoreDestroyResponse = {
    val src = Deepspeed.StoreDestroyResponse.parseFrom(byteString)
    RendezvousStoreDestroyResponse(
      success = src.getSuccess
    )
  }
}

File: jvm/new_core/src/main/java/com/eggroll/core/deepspeed/store/RendezvousStoreDestroyResponse.java
package com.eggroll.core.deepspeed.store;

import com.eggroll.core.pojo.RpcMessage;
import com.google.protobuf.InvalidProtocolBufferException;
import com.webank.eggroll.core.meta.Deepspeed;
import lombok.Data;

@Data
public class RendezvousStoreDestroyResponse implements RpcMessage {
    private boolean success;

    public static byte[] serialize(RendezvousStoreDestroyResponse src) {
        Deepspeed.StoreDestroyResponse.Builder builder = Deepspeed.StoreDestroyResponse.newBuilder()
                .setSuccess(src.isSuccess());
        return builder.build().toByteArray();
    }

    @Override
    public byte[] serialize() {
        Deepspeed.StoreDestroyResponse.Builder builder = Deepspeed.StoreDestroyResponse.newBuilder()
                .setSuccess(this.isSuccess());
        return builder.build().toByteArray();
    }

    @Override
    public void deserialize(byte[] data){
        try {
            Deepspeed.StoreDestroyResponse src = Deepspeed.StoreDestroyResponse.parseFrom(data);
            this.success = src.getSuccess();
        } catch (InvalidProtocolBufferException e) {
            e.printStackTrace();
        }
    }
}